<h1>Find an Equilateral Triangle with conditions</h1>
<font size="4">
Given a point, $P$ in the plane, find an equilateral triangle, $ABC,$ which has $P$ inside, and where $P$ is 3 units from $A$, 2 units from $B$ and 4 units from $C.$ Here I have drawn an approximation to the problem. The point $P=(6,4).$
    
<img src="../Images/Triangle Equilateral Point Inside.png" width="40%">
We will first try to get the side length of the equilateral triangle.  To do that, I will rotate point $P$ about $A$ by 60 degrees and name it $X.$  In doing this, I have created a new equilateral triangle (APX) and, if I connect $X$ with $B$ then a second triangle (PBX).  Their side lengths are known because they were created by $60^\circ$ rotation of known side lengths.
    
<img src="../Images/Triangle Equilateral Point Inside01.png" width="40%">
Next, we can do this again, but this time rotate $P$ by $60^\circ$ about point $B.$ and create $Y.$ and then connect $Y$ and $C$ to made another 2,3,4 triangle.
 
<img src="../Images/Triangle Equilateral Point Inside02.png" width="40%">
In the final instance, we will rotate point $P$ about $C$ by $60^\circ$ and create $Z$ followed by connecting $Z$ and $A$ to the the final 2,3,4 triangle.
    

<img src="../Images/Triangle Equilateral Point Inside03.png" width="40%">
So it turns out that these 6 colored triangles, with all sides known, have exactly twice the area of triangle ABC and that will let us find the area of equilateral triangle ABC and thus its side length.
    
Aside: Why would that be? If we look inside ABC we will find three triangles which are congruent to the same three triangles in the Hexagon that are outside of ABC.  Specifically: $$APC \cong AXB$$
 $$CPB \cong CZA$$  $$APB \cong CYB$$
So, the Hexagon area is twice the Equilateral Triangle Area.
   
    
The (2,3,4) triangle areas can be found from Heron's theorem. $$s_p=\frac{2+3+4}{2}$$<br><br>
$$Area_{234}=\sqrt{s_p(s_p-2)(s_p-3)(s_p-4)}$$ Since $s_p=\frac{9}{2},$ $$Area_{234}=\frac{3\sqrt{15}}{4}$$
We need to remember that there are three of these.
    
Now we can look at the area for each of the three equilateral triangles.
$$Area_{222}=\frac{1}{2}\cdot h \cdot base = \frac{1}{2}\left(\frac{\sqrt{3}}{2}\cdot 2\right)\cdot 2 = \sqrt{3}$$
$$Area_{333}=\frac{1}{2}\cdot h \cdot base = \frac{1}{2}\left(\frac{\sqrt{3}}{2}\cdot 3\right)\cdot 3 = \frac{9\sqrt{3}}{4}$$
$$Area_{444}=\frac{1}{2}\cdot h \cdot base = \frac{1}{2}\left(\frac{\sqrt{3}}{2}\cdot 4\right)\cdot 4  = 4\sqrt{3}$$
Now we sum the areas of the six colored triangles.
$$\text{Hexagon Area} = 3\cdot\left(3 \frac{\sqrt{15}}{4}\right)+\sqrt{3}+\frac{9\sqrt{3}}{4}+4\sqrt{3} = \frac{29\sqrt{3}+9\sqrt{15}}{4}$$

    
We know that the Hexagon area is twice the area of ABC and we know how to calculate the area of an equilateral triangle from its side length.  Let S be the side length of ABC, and Area_{ABC} be it area.
 $$Area_{ABC}=\frac{1}{2}h\cdot S = \frac{1}{2}\frac{\sqrt{3}}{2}S\cdot S$$
 $$Area_{ABC}=\frac{1}{2}\frac{\sqrt{3}}{2}S^2 = \frac{1}{2}\left(\text{Hexagon Area}  \right)$$
 $\therefore$ we can compute S.
 $$\frac{1}{2}\frac{\sqrt{3}}{2}S^2= \frac{1}{2}\frac{29\sqrt{3}+9\sqrt{15}}{4}$$
$$S = \frac{\sqrt{18\sqrt{5}+58}}{2}\approx 4.95603731813523$$
    
 <hr style="border: 1px green solid;">
   

Now we can build the equilateral triangle and put an appropriate radius circle around $A$ and $B$.  It doesn't matter where exactly $A$ and $B$ are located because we will translate the triangle by an appropriate amount to get point P at (6,4).  Use
$$A = (0,0)$$
$$B =\left(x_A+S,y_A+0  \right)$$
$$C = A + S\cdot\left(\begin{array}{c}\cos(60^\circ)\\sin(60^\circ) \end{array}\right)$$

To find the point, $P,$ we intersect the circle around $A$ that has radius $3$  with the circle around $C$ that has radius $2.$  <br>
For the circle about $A,$ $r=3, (h,k)=(0,0)$<br>
For the circle about $B$, $r=2, (h,k)=(S,0)$<br>
From the intersection of two circles, we have the equation of a line that connects the two intersection points. Solve it for x and substitute into either circle to find y:
$$-2x(h_1-h_2)-2y(k_1-k_2)=(r_1^2-r_2^2)-(h_1^2-h_2^2)-(k_1^2-k_2^2)$$

In [38]:
from sympy import *
init_printing(use_latex='mathjax')
var('x y');
S = (sqrt(18*sqrt(5)+58))/2
r1 = 3
r2 = 2
h1 = 0
k1 = 0
h2 = S
k2 = 0
eq1=-2*x*(h1-h2)-2*y*(k1-k2)-(r1**2-r2**2)+(h1**2-h2**2)+(k1**2-k2**2)
soln = solve([eq1], [x])
key, val = next(iter(soln.items()))
display(soln)
print(N(val))
print(val)

x = val
cir1 = (x-h1)**2+(y-k1)**2-r1**2
soln1 = solve(cir1,y)
display(soln1)
print(N(soln1[1]))


⎧             ____________         ____________⎫
⎪     45⋅√5⋅╲╱ 18⋅√5 + 58    363⋅╲╱ 18⋅√5 + 58 ⎪
⎨x: - ──────────────────── + ──────────────────⎬
⎪             872                   872        ⎪
⎩                                              ⎭

2.98245392448661
-45*sqrt(5)*sqrt(18*sqrt(5) + 58)/872 + 363*sqrt(18*sqrt(5) + 58)/872


⎡     _________________       _________________⎤
⎢-3⋅╲╱ 10791 - 4578⋅√5    3⋅╲╱ 10791 - 4578⋅√5 ⎥
⎢───────────────────────, ─────────────────────⎥
⎣          218                     218         ⎦

0.323988562011690


<font size = "4">
So now we have values for point $P^\prime$ whenever the equilateral triangle is level, meaning $A$ and $B$  have the same y coordinate and $A$ is at (0,0).  Therefore, the point $P^\prime$ is used to get the vector from $P^\prime$ to $P.$
  $$P = (6,4)$$<br><br>
  $$P^\prime = \left(\frac{(-45\sqrt{5}\sqrt{18\sqrt{5} + 58}}{872} + \frac{363\sqrt{18\sqrt{5} + 58}}{872},  \frac{3\sqrt{10791-4578\sqrt{5}}}{218}  \right)$$

$$\mathbf{u}=\left(6 - \frac{(-45\sqrt{5}\sqrt{18\sqrt{5} + 58}}{872} + \frac{363\sqrt{18\sqrt{5} + 58}}{872},  4 - \frac{3\sqrt{10791-4578\sqrt{5}}}{218}  \right)$$
The translation of points $A,B,C$ are shown below.
$$A=(0,0)+\mathbf{u}$$
$$B=\left( \frac{\sqrt{18\sqrt{5}+58}}{2},0\right)+\mathbf{u}$$
$$C = A+S\cdot\left(\begin{array}{c}\cos(60^\circ)\\ \sin(60^\circ) \end{array}    \right)$$